Material Design Makeover

ds84182 opened the pull request:
- [x] Partial Material Design page reformatting
- [ ] Primary and Accent colors, pick from [The Material Design Color Palettes please](www.google.com/design/spec/style/color.html#color-ui-color-palette)
- [ ] Animations(ish)

All this absolute positioning nonsense is getting to me :(

List of changes:
Only body should be styled with the background color. html cannot be styled, and giving other elements the same background color only makes the browser's renderer draw more pixels over pixels that it doesn't need.

Moved the submit button to the right of the page. If you look at most websites, the submit buttons are to the right, and configuration is to the left.

wewladdy, those borders are literally Satan. I gave them a little border-radius: 2px; treatment.
# powerof4paddings5lyfe

Many of these prefixed transforms and transitions don't need to exist because transforms and transitions were officially merged into HTML5, making the vendor prefixes a thing of the past. Unless someone is browsing to cpaste on an age old browser (which is already not recomemded by Google), we shouldn't worry about all this vendor prefix replication.

Last, but not least, I removed the child specific selectors. Most to all of the times, web browsers will cache a list of existing ids (since there is only one element with a single id on a page at all times), so when you ask for an id under a certain parent, it will search for all of those parents first, then for the id, instead of looking into the id list for the id that already exists. Since this a relatively simple page, that amount of complexity isn't really needed either.

ds84182 committed:
Some (incomplete) design changes

* [x] Partial Material Design page reformatting
* [ ] Primary and Accent colors, pick from [The Material Design Color Palettes please](www.google.com/design/spec/style/color.html#color-ui-color-palette)

All this absolute positioning nonsense is getting to me :(

ds84182 committed:
Very spooky button:active

background-color: #666; for best color?

ds84182 committed:
Fade in the results link

Fancy.

ds84182 committed:
Positioning specifically for #submit and not for all potential buttons

ds84182 committed:
Use margin of 4px for buttons

ds84182 committed:
Close button

This is possibly the best close button in the world.

ds84182 committed:
Stop W3CValidator's bitching

This also fixes the close button not showing up on ALL browsers.

ChickenNuggers commented:
A few notes:

 * There's no "fade >in<" to the results link - although it does fade out https://github.com/ds84182/cpaste/commit/8b43a28ea48e48ec48f156eaa318dbcf60a39970.
 * The radio buttons aren't really designed at all.
 * The background should be white, if we're planning on going for Material Design. This helps with shadow elements that provide one of the most important elements - Shadow. Once we convert to a white background we should also choose a somewhat bright primary color to use for the buttons to contrast against the shadow.

I realized you said this is incomplete, but this is just making a small To-Do list to finish this up. If you have anything else, feel free to comment.

ds84182 commented:
... It's incomplete and I was requested by @vifino to do a dark theme. And also, Material Design themes can be dark, see [the color page, which discusses text color choices on dark backgrounds, aka dark background applications](http://www.google.com/design/spec/style/color.html#color-color-palette).

ChickenNuggers commented:
Then, if we're making a dark-themed application, I assume we will be changing the background to #030303 instead of #010101 ? Also, during this change, the primary and accent should be applied because some button attributes change it to #030303.

ds84182 committed:
Match color palette with Material Design's grey colors

ds84182 commented:
@vifino and I are still deciding the primary and accent colors at this time. I just went and used MD's actual grey colors for the greys in the style.

ds84182 committed:
Accent is now Deep Purple

I decided to use the lower range (non Axxx section) of Deep Purple because I decided that it matches the dark application more. Material Design is a bit lenient in almost all areas.

ds84182 committed:
Change textarea color to a darker #111111

ds84182 committed:
Fix Firefox bug with textarea

Firefox did not resize the element when both top and bottom were set. However, Chrome does this correctly. Screw you, Firefox.

ds84182 committed:
Flexbox positioning

This throws out the old absolute positioning method (which proves harmful because you don't know how clients are going to render the page at all). A problem I observed in Firefox was that my font was different (Roboto) than whatever the hell Chrome uses, which made the page look different because of the differences in pixel sizes of the fonts.

ds84182 committed:
Fix the god damn insane radio button positioning

Inside, I am crying. Goodness gracious, this crap is too complex. reasonsIHateCSS.add(this);

vifino commented:
Awesome job, @ds84182.

ChickenNuggers commented:
I don't see any problems. ++

ChickenNuggers commented:
Is this done?

vifino commented:
Merged manually.

It's not pretty, but allows me to properly merge it with multi-theme support.
